src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs

```diff
@@ -108,34 +108,65 @@ public void AddCyclicProjectReferencesDeep()
+                project1.SetBinOutputPathAndRelatedData(@"c:\project1.dll");
 
                 var project2 = CreateCSharpProject(environment, "project2");
-                environment.ProjectTracker.UpdateProjectBinPath(project2, null, @"c:\project2.dll");
+                project2.SetBinOutputPathAndRelatedData(@"c:\project2.dll");
 
                 // since this is known to be the output path of project1, the metadata reference is converted to a project reference
                 project2.OnImportAdded(@"c:\project1.dll", "project1");
 
-                Assert.Equal(true, project2.GetCurrentProjectReferences().Any(pr => pr.ProjectId == project1.Id));
+                Assert.Single(project2.GetCurrentProjectReferences().Where(pr => pr.ProjectId == project1.Id));
 
                 // update bin bath for project1.
-                environment.ProjectTracker.UpdateProjectBinPath(project1, @"c:\project1.dll", @"c:\new_project1.dll");
+                project1.SetBinOutputPathAndRelatedData(@"c:\new_project1.dll");
 
                 // Verify project reference updated after bin path change.
-                Assert.Equal(true, project2.GetCurrentProjectReferences().Any(pr => pr.ProjectId == project1.Id));
+                Assert.Empty(project2.GetCurrentProjectReferences());
+
+                // This is a metadata reference to the original path
+                var metadataReference = Assert.Single(project2.GetCurrentMetadataReferences());
+                Assert.Equal(@"c:\project1.dll", metadataReference.FilePath);
 
                 project2.Disconnect();
                 project1.Disconnect();
             }
         }
 
+        [WorkItem(12707, "https://github.com/dotnet/roslyn/issues/12707")]
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void DisconnectingProjectShouldConvertConvertedReferencesBack()
+        {
+            using (var environment = new TestEnvironment())
+            {
+                var project1 = CreateCSharpProject(environment, "project1");
+                project1.SetBinOutputPathAndRelatedData(@"c:\project1.dll");
+
+                var project2 = CreateCSharpProject(environment, "project2");
+                project2.SetBinOutputPathAndRelatedData(@"c:\project2.dll");
+
+                // since this is known to be the output path of project1, the metadata reference is converted to a project reference
+                project2.OnImportAdded(@"c:\project1.dll", "project1");
+
+                Assert.Single(project2.GetCurrentProjectReferences().Where(pr => pr.ProjectId == project1.Id));
+
+                project1.Disconnect();
+
+                // Verify project reference updated after bin path change.
+                Assert.Empty(project2.GetCurrentProjectReferences());
+                Assert.Single(project2.GetCurrentMetadataReferences().Where(r => r.FilePath == @"c:\project1.dll"));
+
+                project2.Disconnect();
```

Comment:
here we probably want to create project1 again and add it back? and see whether projec2 gets that as p2p reference?

Reply:
I think that's already covered in `AddingReferenceToProjectMetadataPromotesToProjectReference` and `AddingProjectReferenceAndUpdateReferenceBinPath` tests effectively?

Reply:
I see. good then.